### PR TITLE
DOC: add links to polynomial function/class listing

### PR DESCRIPTION
**Pull request by madphysicist**
I found these convenient to have on hand while reading.

### CODE REVIEW

File: numpy/polynomial/__init__.py [hunk 1]
```diff
@@ -31,10 +31,10 @@
...
 package, and are available from the `numpy.polynomial` namespace.
 This eliminates the need to
-navigate to the corresponding submodules, e.g. ``np.polynomial.Polynomial``
-or ``np.polynomial.Chebyshev`` instead of
-``np.polynomial.polynomial.Polynomial`` or 
-``np.polynomial.chebyshev.Chebyshev``, respectively.
+navigate to the corresponding submodules, e.g. `numpy.polynomial.Polynomial`
+or `numpy.polynomial.Chebyshev` instead of
+`numpy.polynomial.polynomial.Polynomial` or 
+`numpy.polynomial.chebyshev.Chebyshev`, respectively.
```
Comment (eric-wieser) [inline numpy/polynomial/__init__.py:37]: Do all four of these links actually work in the docs? I'd expect only two of them to work.

Comment (madphysicist) [reply numpy/polynomial/__init__.py:37]: Do any of the CI tests build docs that can be viewed?

Comment (eric-wieser) [reply numpy/polynomial/__init__.py:37]: Yes, here's the broken links as predicted: https://18124-908607-gh.circle-artifacts.com/0/doc/build/html/reference/routines.polynomials.package.html?highlight=polynomial#module-numpy.polynomial

**Comment by eric-wieser** [comment]
Wouldn't it make more sense to put the links in the table above these paragraphs?

**Comment by madphysicist** [comment]
@eric-wieser Excellent idea. Looks like it worked: https://18129-908607-gh.circle-artifacts.com/0/doc/build/html/reference/routines.polynomials.package.html#module-numpy.polynomial

File: numpy/polynomial/__init__.py [hunk 2]
```diff
@@ -58,8 +58,8 @@
 The following lists the various constants and methods common to all of
 the classes representing the various kinds of polynomials. In the following,
 the term ``Poly`` represents any one of the convenience classes (e.g.
-``Polynomial``, ``Chebyshev``, ``Hermite``, etc.) while the lowercase ``p``
-represents an **instance** of a polynomial class.
+`~polynomial.Polynomial`, `~chebyshev.Chebyshev`, `~hermite.Hermite`, etc.)
+while the lowercase ``p`` represents an **instance** of a polynomial class.
```
Comment (eric-wieser) [inline numpy/polynomial/__init__.py:62]: I think I'd be inclined to revert the changes to these lines

File: numpy/polynomial/__init__.py [hunk 3]
```diff
@@ -15,26 +15,26 @@
...
 package, and are available from the `numpy.polynomial` namespace.
 This eliminates the need to
-navigate to the corresponding submodules, e.g. ``np.polynomial.Polynomial``
-or ``np.polynomial.Chebyshev`` instead of
-``np.polynomial.polynomial.Polynomial`` or 
-``np.polynomial.chebyshev.Chebyshev``, respectively.
+navigate to the corresponding submodules, e.g. ``numpy.polynomial.Polynomial``
+or ``numpy.polynomial.Chebyshev`` instead of
+`numpy.polynomial.polynomial.Polynomial` or 
+`numpy.polynomial.chebyshev.Chebyshev`, respectively.
```
Comment (eric-wieser) [inline numpy/polynomial/__init__.py:37]: And to these ones.

File: numpy/polynomial/__init__.py (hunk 2 above)
Comment (madphysicist) [reply numpy/polynomial/__init__.py:62]: It was actually those cases that I wanted to click on. Why not make them links, given that the text did not change?

File: numpy/polynomial/__init__.py (hunk 3 above)
Comment (madphysicist) [reply numpy/polynomial/__init__.py:37]: Same thing. I can make the link text read `np....` instead of `numpy....`, as before, if you prefer, but I think having them as links is quite handy, especially for readers like me that want all the background info up front.

**Comment by madphysicist** [comment]
@eric-wieser I think this is done, unless you really want me to revert those in-test links.

**Comment by charris** [comment]
@eric-wieser Ping.

File: numpy/polynomial/__init__.py (hunk 2 above)
Comment (mattip) [reply numpy/polynomial/__init__.py:62]: I think the intention is that only the first appearance of a keyword should be a link, or maximum the first appearance in each section. Making every appearance a link is distracting.

Comment (madphysicist) [reply numpy/polynomial/__init__.py:61]: @mattip Point taken. I think I've removed all the undesirable links. Please let me know if I missed any.

File: numpy/polynomial/__init__.py [hunk 4]
```diff
@@ -15,26 +15,25 @@
...
-This eliminates the need to
-navigate to the corresponding submodules, e.g. ``np.polynomial.Polynomial``
-or ``np.polynomial.Chebyshev`` instead of
-``np.polynomial.polynomial.Polynomial`` or 
-``np.polynomial.chebyshev.Chebyshev``, respectively.
+package, and are available from the ``numpy.polynomial`` namespace.
+This eliminates the need to navigate to the corresponding submodules, e.g.
+``numpy.polynomial.Polynomial`` or ``numpy.polynomial.Chebyshev`` instead of
+``numpy.polynomial.polynomial.Polynomial`` or
+``numpy.polynomial.chebyshev.Chebyshev``, respectively.
```
Comment (eric-wieser) [inline numpy/polynomial/__init__.py:36]: I don't think changing `np` to `numpy` is an improvement here

Comment (madphysicist) [reply numpy/polynomial/__init__.py]: Agreed. That was to make it a link. Now reverted.

File: numpy/polynomial/__init__.py [hunk 5]
```diff
@@ -15,25 +15,24 @@
...
 The convenience classes are the preferred interface for the `~numpy.polynomial`
-package, and are available from the `numpy.polynomial` namespace.
-This eliminates the need to
-navigate to the corresponding submodules, e.g. ``np.polynomial.Polynomial``
-or ``np.polynomial.Chebyshev`` instead of
-``np.polynomial.polynomial.Polynomial`` or 
+package, and are available from the ``numpy.polynomial`` namespace.
+This eliminates the need to navigate to the corresponding submodules, e.g.
+``np.polynomial.Polynomial`` or ``np.polynomial.Chebyshev`` instead of
+``np.polynomial.polynomial.Polynomial`` or
```
Comment (eric-wieser) [inline numpy/polynomial/__init__.py:35]: Has anything actually changed here?

Comment (madphysicist) [reply numpy/polynomial/__init__.py:35]: No, just fixed the awkward spacing.

File: numpy/polynomial/__init__.py (hunk 2 above)
Comment (rossbar) [reply numpy/polynomial/__init__.py:61]: The original intent here was to highlight the naming convention used for the remainder of the module docstring - i.e. that ``Poly`` will be used as a stand-in for any one of the 6 convenience classes in the following listings.

While the `~<module>.<class>`-style linking works fine in the rendered html, I think it makes the source rst harder to parse in this instance. I tend to agree with @eric-wieser & @mattip that I prefer the original, and having the links to the convenience class in the table that appears at the top makes the most sense.

**Review by rgommers — APPROVED** [review]
LGTM, thanks @madphysicist. And thanks @eric-wieser for reviewing.
